tests/suites/storage/charm_storage.sh

```diff
@@ -0,0 +1,146 @@
+run_charm_storage() {
+	echo
+
+	model_name="test-charm-storage"
+	file="${TEST_DIR}/${model_name}.log"
+
+	ensure "${model_name}" "${file}"
+
+	echo "Assess create-storage-pool"
+	juju create-storage-pool -m "${model_name}" loopy loop size=1G
+	juju create-storage-pool -m "${model_name}" rooty rootfs size=1G
+	juju create-storage-pool -m "${model_name}" tempy tmpfs size=1G
+	juju create-storage-pool -m "${model_name}" ebsy ebs size=1G
+	echo "create-storage-pool PASSED"
+
+	# Assess the above created storage pools.
+	echo "Assessing storage pool"
+	if [ "${BOOTSTRAP_PROVIDER:-}" == "ec2" ]; then
+		juju list-storage-pools -m "${model_name}" --format json | jq '.ebs | .provider' | check "ebs"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.["ebs-ssd"] | .provider' | check "ebs"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.tmpfs | .provider' | check "tmpfs"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.loop | .provider' | check "loop"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.rootfs | .provider' | check "rootfs"
+	else
+		juju list-storage-pools -m "${model_name}" --format json | jq '.rooty | .provider' | check "rootfs"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.tempy | .provider' | check "tmpfs"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.loopy | .provider' | check "loop"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.ebsy | .provider' | check "ebs"
+	fi
+	echo "Storage pool PASSED"
+
+	# Assess charm storage with the filesystem storage provider
+	echo "Assessing filesystem rootfs"
+	juju deploy ./testcharms/charms/dummy-storage-fs --series="jammy" --storage data=rootfs,1G
+	wait_for "dummy-storage-fs" ".applications"
+	# assert the storage kind name
+	juju list-storage --format json | jq '.storage[] | .kind' | check "filesystem"
+	# assert the storage label name
+	juju list-storage --format json | jq '.storage | keys | .[]' | check "data/0"
+	# assert the unit attachment name
+	juju list-storage --format json | jq '.storage[] | .attachments | .units | keys | .[]' | check "dummy-storage-fs/0"
+	# assert the attached unit state
+	juju list-storage --format json | jq '.storage[] | .attachments | .units[] | .life' | check "alive"
+	echo "Filesystem rootfs PASSED"
+	# remove the application
+	juju remove-application dummy-storage-fs
+	wait_for "{}" ".applications"
```

Comment:
Curious to know how you would deploy charms in parallel using bash.

Reply:
The only advantage is it would save you a bit of time, which isn't too important since I don't think this test takes particularly long. If cleaning up between parts is more convenient this would be fine by me

I was thinking something like:
```
juju deploy dummy-storage-fs ...
juju deploy dummy-storage-lp ...
juju deploy dummy-storage-tp ...

wait_for dummy-storage-fs  ${idle_condition dummy-storage-fs}
... # test dummy-storage-fs
...

wait_for dummy-storage-lp  ${idle_condition dummy-storage-lp}
... # test dummy-storage-lp
...

wait_for dummy-storage-tp  ${idle_condition dummy-storage-tp}
... # test dummy-storage-tp
...
```

Reply:
Looks like the storage unit is incremented weather or not you destroy the previous storage btw

Reply:
I have refactored to a address this, it has resulted to a major refactor of the test assertions as it resulted in different storage unit indices but the storage app indices i.e `dummy-storage-lp/0` remain the same.